ddtrace/propagation/http.py

```diff
@@ -950,6 +966,18 @@ def parent_call():
             for key in span_context._baggage:
                 headers[_HTTP_BAGGAGE_PREFIX + key] = span_context._baggage[key]
 
+        if hasattr(ddtrace, "tracer") and hasattr(ddtrace.tracer, "sample"):
+            if non_active_span is not None:
+                if _is_top_level(non_active_span):
```

Comment:
top level will return True for service entry spans. Shouldn't we only sample on root spans

Reply:
If you look down a few lines, we check `if root_span is not None and root_span.context.sampling_priority is None` which essentially makes sure we're only sampling on a root span.

Reply:
I did change the logic a bit though, to just grab span._local_root, since if you look at the start_span logic, we always set the span._local_root to the span, if there's no parent for it.